### PR TITLE
Address SA feedback

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/core/881?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `feature/proof-of-stake`
  * **PR #878** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/878?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #881** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/881?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/881?utm_source=stack-comment).

File: lib/pos_mempool.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ const (
 )
 
 type Mempool interface {
-	Start() error
+	Start(feeEstimatorPastBlocks []*MsgDeSoBlock, feeEstimatorNumBlocks uint64) error
```
Comment (tholonious) [inline lib/pos_mempool.go:22]: WDYT about having a separate Init(...) function, which would let you keep the Start() function scoped to just starting the mempool's event loop? Ideally, we're able to call Start() for a place where we don't have these params available on-hand. Decoupling the two now will open up some more options for us down the line
```suggestion
	Init(feeEstimatorPastBlocks []*MsgDeSoBlock, feeEstimatorNumBlocks uint64) error
	Start() error
```

File: lib/legacy_mempool.go [hunk 2]
```diff
@@ -2422,8 +2422,22 @@ func (mp *DeSoMempool) GetMempoolSummaryStats() (_summaryStatsMap map[string]*Su
 	return convertMempoolTxsToSummaryStats(mp.readOnlyUniversalTransactionList)
 }
 
-func (mp *DeSoMempool) GetFeeEstimator() *PoSFeeEstimator {
-	return nil
+func EstimateFee(txn *MsgDeSoTxn, minFeeRateNanosPerKB uint64) uint64 {
```
Comment (tholonious) [inline lib/legacy_mempool.go:2425]: What's a more self descriptive name for this? Here's one option:
```suggestion
func EstimateMaxTxnFeeV1(txn *MsgDeSoTxn, minFeeRateNanosPerKB uint64) uint64 {
```

File: lib/pos_mempool.go (hunk 1 above)
Comment (lazynina) [reply lib/pos_mempool.go:22]: We can actually move the Init out of the interface I think!

**Review by tholonious — APPROVED** [review]

**Pull request merged**